### ISSUE TITLE
`Generic` instances generation for case classes with varargs

### ISSUE THREAD

**Issue by netvl** [milessabin/shapeless#327]
Currently shapeless can't generate `Generic` instances for case classes with variable arguments:
```
scala> import shapeless._
import shapeless._

scala> case class A(x: Int, ys: String*)
defined class A

scala> Generic[A]
<console>:13: shapeless.this.Generic.materialize is not a valid implicit value for shapeless.Generic[A] because:
hasMatchingSymbol reported error: type mismatch;
 found   : String
 required: String*
              Generic[A]
                     ^
<console>:13: error: could not find implicit value for parameter gen: shapeless.Generic[A]
              Generic[A]
                     ^
```

Maybe it makes sense to treat the last argument as `Seq[_]` or something like that.

**Comment by milessabin** [comment]
Good catch ... do you want to take a pass at it?

**Comment by netvl** [comment]
@milessabin, do you suggest me to try and fix this? I'd like to, but I will need some time in order to understand how the macro machinery works.

**Comment by milessabin** [comment]
Yup.

If no one else picks it up, I will, before the next release, but it might make an interesting project for someone who's interested in shapeless and wants to get a feel for macros.

I'd say not "low hanging fruit" (because macros) but definitely doable.

**Comment by netvl** [comment]
Yeah, count me in then :)

**Comment by milessabin** [comment]
Feel free to assign it to yourself if/when you have time to work on it.

**Comment by netvl** [comment]
@milessabin, I can't assign an issue to myself but anyway, I've submitted a PR.

**Comment by milessabin** [comment]
Fixed in #330.
